src/test/scala/tag/TagDaoTest.scala

```diff
@@ -89,6 +96,115 @@ class TagDaoTest
+          tag <- tagDao.getTag(cid, tid).map(_.get)
+        } yield {
+          affectedRows shouldBe 1
+          tag.name shouldBe description.name
+          tag.color shouldBe description.color
+        }
+      }
+
+      it("should update the name and color") {
+        val newDescription = baseDescription.copy(
+          name = "New Name",
+          color = "#000000"
+        )
+
+        testCreateTag(newDescription)
+      }
+
+      it("should update the name") {
+        testCreateTag(baseDescription.copy(name = "New Name"))
+      }
+
+      it("should update the color") {
+        testCreateTag(baseDescription.copy(color = "#000000"))
+      }
+
+//    describe("updateTag") {
+//      it("should update name and color") {
+//        for {
+//          tid <- tagDao.createTag(tagOne).rightGet()
+//          rows <- tagDao.updateTag(cid, tid, tagDescriptionNew).rightGet()
+//          tag <- tagDao.getTag(cid, tid).map(x => x.get)
+//
+//        } yield {
+//          rows shouldBe 1
+//          tag.name shouldBe tagDescriptionNew.name
+//          tag.color shouldBe tagDescriptionNew.color
+//        }
+//      }
+//
+//      it("should update name") {
+//        for {
+//          tid <- tagDao.createTag(tagOne).rightGet()
+//          rows <- tagDao.updateTag(cid, tid, tagDescriptionNewName).rightGet()
+//          tag <- tagDao.getTag(cid, tid).map(x => x.get)
+//
+//        } yield {
+//          rows shouldBe 1
+//          tag.name shouldBe tagDescriptionNewName.name
+//          tag.color shouldBe tagDescriptionNewName.color
+//        }
+//      }
+//
+//      it("should update color") {
+//        for {
+//          tid <- tagDao.createTag(tagOne).rightGet()
+//          rows <- tagDao.updateTag(cid, tid, tagDescriptionNewColor).rightGet()
+//          tag <- tagDao.getTag(cid, tid).map(x => x.get)
+//
+//        } yield {
+//          rows shouldBe 1
+//          tag.name shouldBe tagDescriptionNewColor.name
+//          tag.color shouldBe tagDescriptionNewColor.color
+//        }
+//      }
+
+      it("should fail for duplicate compound cid + name") {
+        for {
+          tid <- tagDao.createTag(tagOne).rightGet()
+          _ <- tagDao.createTag(tagTwo).rightGet()
+          errorOrTag = tagDao
+            .updateTag(cid, tid, tagDescriptionDuplicate)
+            .unsafeRun()
+        } yield {
+          errorOrTag shouldBe Left(
+            DuplicateKey(
+              f"Tag with name ${tagTwo.name} already exists for CID ${tagTwo.cid}."
+            )
+          )
+        }
+      }
+
+      it("should fail for non existing tag") {
+        for {
+          _ <- tagDao.createTag(tagOne).rightGet()
+          rows <- tagDao.updateTag(cid, 99, tagDescriptionNewColor).rightGet()
+        } yield {
+          rows shouldBe 0
+        }
+      }
+      it("should fail for tag not owned by the cid") {
+        for {
+          _ <- tagDao.createTag(tagOne).rightGet()
```

Comment:
```suggestion
          _ <- tagDao.createTag(tagOne)
```
no need for `rightGet`